cluster-autoscaler/cloudprovider/baiducloud/baiducloud-sdk-go/cce/cluster.go

```diff
@@ -348,14 +370,138 @@ func (c *Client) ScaleDownCluster(args *ScaleDownClusterArgs) error {
 			"scalingDown": "",
 		}
 	}
+
+	klog.Infof("ScaleDownCluster args: %v", params)
 	postContent, err := json.Marshal(args)
 	if err != nil {
 		return err
 	}
 	req, err := bce.NewRequest("POST", c.GetURL("v1/cluster", params), bytes.NewBuffer(postContent))
+	klog.Infof("ScaleDownCluster req: %v", req)
+
 	if err != nil {
 		return err
 	}
 	_, err = c.SendRequest(req, nil)
 	return err
 }
+
+// DescribeGroup returns the description of the group
+func (c *Client) DescribeGroup(groupID string, clusterID string) (*CceGroup, error) {
+	if clusterID == "" {
+		return nil, fmt.Errorf("clusterID should not be nil")
+	}
+	if groupID == "" {
+		return nil, fmt.Errorf("groupID should not be nil")
+	}
+
+	params := map[string]string{
+		"clusterUuid": clusterID,
+		"groupId":     groupID,
+	}
+	req, err := bce.NewRequest("GET", c.GetURL("/v1/cluster/group", params), nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.SendRequest(req, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	bodyContent, err := resp.GetBodyContent()
+
+	if err != nil {
+		return nil, err
+	}
+
+	var cceGroup CceGroup
+	err = json.Unmarshal(bodyContent, &cceGroup)
+
+	if err != nil {
+		return nil, err
+	}
+	return &cceGroup, nil
+}
+
+// GetAsgNodes returns the group's nodes
+func (c *Client) GetAsgNodes(groupID string, clusterID string) ([]CceInstance, error) {
+	if clusterID == "" {
+		return nil, fmt.Errorf("clusterID should not be nil")
+	}
+
+	if groupID == "" {
+		return nil, fmt.Errorf("groupID should not be nil")
+	}
+
+	params := map[string]string{
+		"clusterUuid": clusterID,
+		"groupId":     groupID,
+	}
+	req, err := bce.NewRequest("GET", c.GetURL("/v1/cluster/group/instances", params), nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := c.SendRequest(req, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	bodyContent, err := resp.GetBodyContent()
+
+	if err != nil {
+		return nil, err
+	}
+
+	var insList ListInstancesResponse
+	err = json.Unmarshal(bodyContent, &insList)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return insList.Instances, nil
+}
+
+// ScaleUpClusterWithGroupID scales up cluster
+func (c *Client) ScaleUpClusterWithGroupID(args *ScaleUpClusterWithGroupIDArgs) error {
+	if args == nil || args.ClusterID == "" ||
+		args.GroupID == "" || args.Num < 0 {
+		return fmt.Errorf("ScaleUpClusterWithGroupIDArgs err")
+	}
+
+	var params map[string]string
+	if args != nil {
+		params = map[string]string{
+			"groupId":     args.GroupID,
+			"clusterUuid": args.ClusterID,
+			"num":         strconv.Itoa(args.Num),
+		}
+	}
+
+	req, err := bce.NewRequest("POST", c.GetURL("v1/cluster/group/scaling_up", params), nil)
+	if err != nil {
+		return err
+	}
+	resp, err := c.SendRequest(req, nil)
+	if err != nil {
+		return err
+	}
+	bodyContent, err := resp.GetBodyContent()
+	if err != nil {
+		return err
+	}
+	var scResp *ScaleUpClusterResponse
+	err = json.Unmarshal(bodyContent, &scResp)
```

Comment:
take address of pointer? a bug?

Reply:
Yes, but does not affect the function. I will fix it Immediately.